examples/peft_llm_gnn/llm_gnn_model.py

```diff
@@ -0,0 +1,142 @@
+"""
+    Copyright 2023 Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    LLM-GNNs implementation.
+"""
+import dgl
+import graphstorm as gs
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import os
+
+from graphstorm import model as gsmodel
+from graphstorm.model.lm_model import TOKEN_IDX, ATT_MASK_IDX
+
+from peft import LoraConfig, get_peft_model
+from transformers import (
+    AutoConfig,
+    AutoModel,
+)
+
+def get_lm_node_feats(g, lm_feat_names, lm_ntypes):
+    """ Collect language model related node features
+
+    Parameters
+    ----------
+    g: graph
+        Graph data.
+    lm_feat_names: list of str
+        A list of llm features
+    lm_ntypes: list of str
+        A list of node types.
+
+    Return
+    ------
+    A dict of dict of distributed tensor.
+        {Node type: {Feature name: Feature stored as distributed tensor}}
+    """
+    lm_feats = {}
+    for ntype in lm_ntypes:
+        lm_feats[ntype] = {}
+        for lm_fname in lm_feat_names:
+            if lm_fname in g.nodes[ntype].data:
+                lm_feats[ntype][lm_fname] = g.nodes[ntype].data[lm_fname]
+
+    return lm_feats
+
+
+class LLMGraphModel(gsmodel.GSgnnNodeModelBase):
+    """ LLMGNN Model Class
+
+    Parameters
+    ----------
+    """
+    def __init__(self, g, node_lm_configs, h_dim, out_dim, num_layers,
+                 target_ntype,     # the node type to be predict
+                 use_norm = True,   # use normalization or not, default is True
+                 alpha_l2norm = 0,
+                 lr = 0.001):
+        super(LLMGraphModel, self).__init__()
+        self.num_layers = num_layers
+        self.target_ntype=target_ntype
+        self.alpha_l2norm = alpha_l2norm
+        self.lr = lr
+        # assume only one LLM is used
```

Comment:
I think it's better to add a comment to explain that only one LLM is supported. also, we should verify that a user only specify one LLM. if a user specifies more than one LLM, we should print a warning.